fdbcli/DebugCommands.actor.cpp

```diff
@@ -164,11 +164,12 @@ ACTOR Future<bool> getallCommandActor(Database cx, std::vector<StringRef> tokens
 CommandFactory getallCommandFactory("getall");
 
 // check that all replies are the same. Update begin to the next key to check
-bool checkResults(Version version,
-                  const std::vector<StorageServerInterface>& servers,
-                  const std::vector<Future<ErrorOr<GetKeyValuesReply>>>& replies,
-                  KeySelectorRef& begin,
-                  const KeySelectorRef& end) {
+std::pair<bool, bool> checkResults(Version version,
+                                   const std::vector<StorageServerInterface>& servers,
+                                   const std::vector<Future<ErrorOr<GetKeyValuesReply>>>& replies,
```

Comment:
A dumm question out of the changes in this RP:

Is it possible that one of the `GetKeyValuesReply`, say from ss0 reads all the 1000 keys in that shard with `.more` as false, but the `GetKeyValuesReply` from `ss1` reads 500 in this round, with `.more` as true?

Reply:
its possible, and `allSame` will be false due to [here](https://github.com/apple/foundationdb/blob/1796b1ffa3f3eb1882e0a29d01183058186febbf/fdbcli/DebugCommands.actor.cpp#L190-L192), and the caller will return accordingly.

Reply:
Will it continue to the following print sections and treat the missing ones as `unique` keys?

Reply:
yes i believe so, it would not `continue` at [here](https://github.com/apple/foundationdb/blob/1796b1ffa3f3eb1882e0a29d01183058186febbf/fdbcli/DebugCommands.actor.cpp#L191), and then proceed to the printing part.